test/unit/mdc-select/mdc-select.test.js

```diff
@@ -209,232 +252,55 @@ test('adapter#deactivateBottomLine removes active class from the bottom line', (
-});
-
-test('adapter#create2dRenderingContext returns a CanvasRenderingContext2d instance', () => {
-  const {component} = setupTest();
-  const fakeCtx = {};
-  const fakeCanvas = {
-    getContext: td.func('canvas.getContext'),
-  };
-  const origCreateElement = document.createElement;
-  document.createElement = td.func('document.createElement');
-
-  td.when(fakeCanvas.getContext('2d')).thenReturn(fakeCtx);
-  td.when(document.createElement('canvas')).thenReturn(fakeCanvas);
-
-  const ctx = component.getDefaultFoundation().adapter_.create2dRenderingContext();
-  assert.equal(ctx, fakeCtx);
-
-  document.createElement = origCreateElement;
-});
-
-test('adapter#setMenuElStyle sets a style property on the menu element', () => {
-  const {component, menuEl} = setupTest();
-  component.getDefaultFoundation().adapter_.setMenuElStyle('font-size', '10px');
-  assert.equal(menuEl.style.fontSize, '10px');
-});
-
-test('adapter#setMenuElAttr sets an attribute on the menu element', () => {
-  const {component, menuEl} = setupTest();
-  component.getDefaultFoundation().adapter_.setMenuElAttr('aria-hidden', 'true');
-  assert.equal(menuEl.getAttribute('aria-hidden'), 'true');
-});
-
-test('adapter#rmMenuElAttr removes an attribute from the menu element', () => {
-  const {component, menuEl} = setupTest();
-  menuEl.setAttribute('aria-hidden', 'true');
-  component.getDefaultFoundation().adapter_.rmMenuElAttr('aria-hidden');
-  assert.isNotOk(menuEl.hasAttribute('aria-hidden'));
-});
-
-test('adapter#getMenuElOffsetHeight returns the menu element\'s offsetHeight', () => {
-  const {component, menuEl} = setupTest();
-  assert.equal(component.getDefaultFoundation().adapter_.getMenuElOffsetHeight(), menuEl.offsetHeight);
-});
-
-test('adapter#openMenu shows the menu with the given focusIndex', () => {
-  const {component, menu} = setupTest();
-  component.getDefaultFoundation().adapter_.openMenu(1);
-  td.verify(menu.show({focusIndex: 1}));
-});
-
-test('adapter#isMenuOpen returns whether or not the menu is open', () => {
-  const {component, menu} = setupTest();
-  const {adapter_: adapter} = component.getDefaultFoundation();
-  menu.open = true;
-  assert.isOk(adapter.isMenuOpen());
-  menu.open = false;
-  assert.isNotOk(adapter.isMenuOpen());
-});
-
-test('adapter#setSelectedTextContent sets the textContent of the selected text el', () => {
-  const {component, fixture} = setupTest();
-  component.getDefaultFoundation().adapter_.setSelectedTextContent('content');
-  assert.equal(fixture.querySelector('.mdc-select__selected-text').textContent, 'content');
-});
-
-test('adapter#getNumberOfOptions returns the length of the component\'s options property', () => {
+test('adapter#getValue returns the nativeControl value', () => {
   const {component} = setupTest();
-  assert.equal(component.getDefaultFoundation().adapter_.getNumberOfOptions(), component.options.length);
+  component.getDefaultFoundation().adapter_.setValue('orange');
```

Comment:
Replace this with `nativeControl.value = 'orange'` to ensure we're actually testing what we say we're testing. This would currently pass if the set/getValue APIs match even if neither one touches the native input.

(You're doing the right thing below to test getSelectedIndex, so not sure why this one was done like this.)